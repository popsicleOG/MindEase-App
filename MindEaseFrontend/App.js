import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import HomeScreen from './src/screens/HomeScreen';
import MoodTrackerScreen from './src/screens/MoodTrackerScreen';
import MindfulnessScreen from './src/screens/MindfulnessScreen';
import CommunityHubScreen from './src/screens/CommunityHubScreen';
import ProfileScreen from './src/screens/ProfileScreen';
import LoginScreen from './src/screens/LoginScreen';

const Tab = createBottomTabNavigator();

const App = () => {
  return (
    <NavigationContainer>
      <Tab.Navigator
        screenOptions={{
          tabBarStyle: { backgroundColor: '#1A3C6E' },
          tabBarActiveTintColor: '#FFFFFF',
          tabBarInactiveTintColor: '#A3BFFA',
        }}
      >
        <Tab.Screen
          name="Login"
          component={LoginScreen}
          options={{ tabBarStyle: { display: 'none' } }}
        />
        <Tab.Screen name="Home" component={HomeScreen} />
        <Tab.Screen name="MoodTracker" component={MoodTrackerScreen} />
        <Tab.Screen name="Mindfulness" component={MindfulnessScreen} />
        <Tab.Screen name="Community" component={CommunityHubScreen} />
        <Tab.Screen name="Profile" component={ProfileScreen} />
      </Tab.Navigator>
    </NavigationContainer>
  );
};

export default App;
