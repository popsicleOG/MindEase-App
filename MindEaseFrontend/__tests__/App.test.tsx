/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import App from '../App';

describe('App Component', () => {
  test('renders without crashing', () => {
    ReactTestRenderer.act(() => {
      const tree = ReactTestRenderer.create(<App />);
      expect(tree).toBeTruthy();
    });
  });

  test('renders with correct structure', () => {
    ReactTestRenderer.act(() => {
      const tree = ReactTestRenderer.create(<App />);
      expect(tree).toBeTruthy();
    });
  });

  test('component snapshot matches', () => {
    ReactTestRenderer.act(() => {
      const tree = ReactTestRenderer.create(<App />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
