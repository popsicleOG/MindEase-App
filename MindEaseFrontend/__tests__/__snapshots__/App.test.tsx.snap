// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component component snapshot matches 1`] = `null`;
